File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        for(int i=0;i<s.length();i++)
        {
            if(s[i]=='('|| s[i]=='{' || s[i]== '[')
            {
                st.push(s[i]);
            }
            else
            {
                if(st.empty())
                {
                    return false;
                    
                }
                else
                {
                    char first=st.top();
                    st.pop();
                    if((first!='('&& s[i]==')') || (first!='{'&& s[i]=='}') || (first!='['&& s[i]==']'))
                        return false;
                    
                }
            }
        }
        return st.empty();
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        stack = []
        parantheses = {')':'(', '}':'{' ,']':'['}

        for i in s:
            if i in '({[':
                stack.append(i)
            elif i in ')}]':
                if not stack or stack[-1] != parantheses[i]:
                    return False
                stack.pop()

        return len(stack)==0
